File: academic-frontend/fix_urls.py
import re

with open('js/dashboard.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Replace single-quoted hardcoded URL with backtick template string
content = re.sub(r"'https://dirictiondback\.digitalrace\.net(.*?)'", r"`${API_BASE}\1`", content)

# Replace already backtick template string hardcoded URL
content = re.sub(r"`https://dirictiondback\.digitalrace\.net(.*?)`", r"`${API_BASE}\1`", content)

# Change API_BASE to localhost for local testing
content = content.replace("const API_BASE = 'http://localhost:8085';", "const API_BASE = 'http://localhost:8080';")

# Change size=30 to size=10
content = content.replace('?size=30', '?size=10').replace('&size=30', '&size=10')

with open('js/dashboard.js', 'w', encoding='utf-8') as f:
    f.write(content)
print("Done")

File: academic-frontend/fix_urls.js
const fs = require('fs');

let content = fs.readFileSync('js/dashboard.js', 'utf8');

// Replace single-quoted hardcoded URL with backtick template string
content = content.replace(/'https:\/\/dirictiondback\.digitalrace\.net([^']*)'/g, "`${API_BASE}$1`");

// Replace already backtick template string hardcoded URL
content = content.replace(/`https:\/\/dirictiondback\.digitalrace\.net([^`]*)`/g, "`${API_BASE}$1`");

// Change API_BASE to localhost for local testing
content = content.replace("const API_BASE = 'http://localhost:8085';", "const API_BASE = 'http://localhost:8080';");

// Change size=30 to size=10
content = content.replaceAll('?size=30', '?size=10').replaceAll('&size=30', '&size=10');

fs.writeFileSync('js/dashboard.js', content, 'utf8');
console.log("Done");
